kernel/src/process_loading.rs

```diff
@@ -699,92 +667,66 @@ impl<C: Chip, D: ProcessStandardDebug> SequentialProcessLoaderMachine<'_, C, D>
                 // doesn't conflict with any of those. Since those processes
                 // are already loaded, we just need to check if this process
                 // binary has the same AppID as an already loaded process.
-                self.procs.map(|procs| {
-                    for proc in procs.iter() {
-                        match proc {
-                            Some(p) => {
-                                let blocked =
-                                    self.is_blocked_from_loading_by_process(&process_binary, *p);
+                for proc in self.kernel.get_process_iter() {
+                    let blocked = self.is_blocked_from_loading_by_process(&process_binary, proc);
 
-                                if blocked {
-                                    ok_to_load = false;
-                                    break;
-                                }
-                            }
-                            None => {}
-                        }
+                    if blocked {
+                        ok_to_load = false;
+                        break;
                     }
-                });
+                }
 
                 if !ok_to_load {
                     continue;
                 }
 
                 // If we get here it is ok to load the process.
-                match self.find_open_process_slot() {
```

Comment:
This is perhaps a notable change. If we don't have anywhere to store the process I think this will use up resources for the process and then just loose track of those resources (ie memory). Whereas before we wouldn't even try to load the process.

Reply:
If `find_open_process_slot()` fails, we will never call `load_process` so do not waste any memory for processes.

if it succeeds, but `load_process` fails, then we do not waste the slot in the array because calling this does not reserve the slot.

The danger behind this interface is do we not reserve this slot, so if someone wrote a strange process loader that nested process loading in itself the pattern would break.

Reply:
I don't understand. This PR removes `find_open_process_slot()`.